client/packages/host/src/Admin/Settings.tsx

```diff
@@ -133,8 +134,9 @@ export const Settings: React.FC = () => {
     const mode =
       nativeMode === NativeMode.Server ? NativeMode.Client : NativeMode.Server;
 
-    localStorage.removeItem(PREVIOUS_SERVER_KEY);
-    setNativeMode(mode);
+    setPreference('previousServer', {} as FrontEndHost);
```

Comment:
technically for async operations, something like below or in a new promise, etc.. not sure if it could be the case, but if setting preferences take longer then navigation would have an issue here

```
    (async () => {
      await setPreference('previousServer', {} as FrontEndHost);
      await setPreference('mode', mode);
      navigate(RouteBuilder.create(AppRoute.Android).build());
    })();
```

Reply:
thanks - that's much better